jujugui/static/gui/src/app/views/topology/relation.js

```diff
@@ -841,15 +837,13 @@ YUI.add('juju-topology-relation', function(Y) {
 
       // Transform endpoints into a list of relatable services (to the
       // service).
-      var possible_relations = Y.Array.map(
-          Y.Array.flatten(Y.Object.values(endpoints)),
-          function(ep) {return ep.service;});
+      var possible_relations = Y.Array.flatten(Y.Object.values(endpoints)).map(
```

Comment:
I was trying to do these by removing each YUI util one by one, especially with some of these where we will need to create our own util.